### ISSUE TITLE
Logical AND and OR for Tensors in C++ API.

### ISSUE THREAD

**Issue by v0dro** [pytorch/pytorch#32614]
## 🚀 Feature

The `at::Tensor` datatype does not have `&` overloaded the way it is in the python API, thus not allowing for boolean element-wise operations between tensors.

## Motivation

I want to call the `at::where()` function with a boolean tensor that is the outcome of the AND of two conditions.

## Pitch

Something like this should be possible:
``` c++
  auto output = at::where((target > -EPSILON) & (target < EPSILON), all_zeros, whatever_values);
```
## Alternatives

This is ugly and tedious IMO:

``` c++
  auto output = at::where(target > -EPSILON, output_pos, zeros);
  output = at::where(output < EPSILON, zeros, output_pos);
```

**Comment by yf225** [comment]
@v0dro Yes it would be awesome to have the logical AND and OR operators for C++ tensors. Thanks a lot for your help!

**Comment by v0dro** [comment]
Can you please tell where exactly the tests for tensor operator tests should reside for C++? I could not find the C++ tests for operators anywhere.

**Comment by yf225** [comment]
@v0dro It would be a great idea to add them to `test/cpp/api/tensor.cpp` :D

**Comment by yf225** [comment]
(Apologies for posting the previous comment too fast)

@v0dro It would be a great idea to create a new test file `aten/src/ATen/test/tensor_operator_test.cpp`, and add the tests there. :D Thanks!
